src/main/java/com/wolox/training/models/Users.java

```diff
@@ -0,0 +1,95 @@
+package com.wolox.training.models;
+
+import com.sun.istack.NotNull;
+import com.wolox.training.exceptions.BookAlreadyOwnedException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+
+@Entity
+public class Users {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
+
+    @Column(nullable = false)
+    @NotNull
+    private String userName;
+
+    @Column(nullable = false)
+    @NotNull
+    private String name;
+
+    @Column(nullable = false)
+    @NotNull
+    private LocalDate birthDate;
+
+    @Column(nullable = false)
+    @NotNull
+    @ManyToMany(cascade = CascadeType.ALL)
```

Comment:
Here you don't want to use CascadeType.ALL. That means (Persist / Delete / Merge / Update / Detach).

Given the users "live/die" independant of the books they may have associated, the only thing you really want here is to keep the Join Table updated. 

Persist and Delete to a User would forward the persist/delete operation applied to it's associated Books. Detach is used to manually detach the items from the persistance context (we never do this manually).

So we're left with Merge and Update. Merge persists memory changes to the database, and refresh brings them from the database to memory. These are the only two we need here :)